modules/dnn/src/onnx/onnx_importer.cpp

```diff
@@ -0,0 +1,275 @@
+// This file is part of OpenCV project.
+// It is subject to the license terms in the LICENSE file found in the top-level directory
+// of this distribution and at http://opencv.org/license.html.
+
+// Copyright (C) 2018, Intel Corporation, all rights reserved.
+// Third party copyrights are property of their respective owners.
+
+#include "../precomp.hpp"
+
+#ifdef HAVE_PROTOBUF
+
+#include <iostream>
+#include <fstream>
+#include <string>
+#include <limits>
+
+
+#if defined(__GNUC__) && __GNUC__ >= 5
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wsuggest-override"
+#endif
+#include "opencv-onnx.pb.h"
+#if defined(__GNUC__) && __GNUC__ >= 5
+#pragma GCC diagnostic pop
+#endif
+
+
+#endif
```

Comment:
Is this "endif" for `HAVE_PROTOBUF`?

This should be moved to the end of file.

Ideally, with "#else" branch with "Non Implemented" stubs for public API (but it is not used yet for caffe).